### DISCUSSION TITLE
Flow confusion -- downstream errors reported as upstream

### DISCUSSION THREAD

**Discussion by gene-hightower** [haraka/Haraka#3131] [General]
**Describe the bug**
There are error messages in outbound/hmail.js that refer to downstream servers as "upstream" servers.

**Expected behavior**
As I understand the flow of streams, it is *from* upstream *to* downstream, in accordance with gravity.

**Comment by baudehlo** [comment]
I picture it like this: External <-> Haraka <-> Internal (e.g. IMAP server).

In that picture, downstream is the IMAP server. Upstream is outbound.

**Reply by gene-hightower** [reply]
Any particular message (unique Message-Id) flows in only one direction.

**Reply by gene-hightower** [reply]
That flow should be shown in the Received: headers added to the message at each hop.
